include/IECoreScene/private/PrimitiveAlgoUtils.h

```diff
@@ -133,61 +133,50 @@ struct AverageValueFromVector
 	}
 };
 
+template< typename T > struct IsVectorDataDefined
+	: boost::mpl::and_
+	<
+		IECore::TypeTraits::IsSimpleTypedData< T >,
+		boost::mpl::not_< IECore::TypeTraits::IsSplineTypedData< T > >,
+		boost::mpl::not_< IECore::TypeTraits::IsTransformationMatrixTypedData< T > >,
+		boost::mpl::not_< IECore::TypeTraits::IsPathMatcherTypedData< T > >,
+		boost::mpl::not_< IECore::TypeTraits::IsDateTimeTypedData< T > >
+	>
+{};
```

Comment:
This seems fragile - it's reasonably likely we'll add other SimpleTypedData in the future, and maintaining the list of `not_` will be a bit of a pain. I think we can do better by using the fact that `TypedDataTraits::DataHolder` is void for undefined data types :

```
template<typename T>
using IsTypedDataDefined = std::negation<std::is_void<typename IECore::TypedDataTraits<T>::DataHolder>>;
```

I've pushed [a commit](https://github.com/johnhaddon/cortex/commit/67cd651a5da70ac75969e532ef3cfd802c061740) to my own branch that demonstrates this working. The other nice thing about this is that we don't need any of the additional functions in `TypeTraits.h`.

Reply:
done